### ISSUE TITLE
[HOLD for payment 2021-11-04] [n6-hold bonus with payment reminder] Report a bug, doesn't work as expected.

### ISSUE THREAD

**Issue by Santhosh-Sellavel** [Expensify/App#4545]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to Settings -> About
2. Click on `Report a bug`
3.`Concierge` chat will be opened now.
4.  Now repeat steps 1 & 2.
5. `Concierge` is not open instead navigate back to the Settings page

For new users it's bit more iritating. They would never know conciegre help. Because the new user will have only concierge active by default. 

## Expected Result:
I believe `Report a bug` should open the `Concierge` chat every time if that's the desired action.

## Actual Result:
It's not opened instead navigated back to the Settings page.

https://user-images.githubusercontent.com/85645967/129031672-d8f6899c-589a-49bd-8090-176693739593.mp4

## Workaround:
Yes, the user could manually close the settings page.

## Suggestions
IMO Just opening the concierge does not look like the user can report bugs in concierge chat. We can show some additional messages like you can start reporting by just adding certain placeholder text in chat input. Ex What can I help you with?, Tell us your problem, etc.

## Platform:
All of the following
Where is this issue occurring?
- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mateocole (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by Santhosh-Sellavel** [comment]
## Proposal
The below lines invoke the action to open the `Concierge` chat.
https://github.com/Expensify/App/blob/83dd58e4dc5f63d97f78cf7f0df434d9b3a47389/src/pages/settings/AboutPage.js#L67-L69

### Simple solution:
We can straightaway call `Navigation.dismissModal()` to dismiss the view, or we could add an optional callback in `fetchOrCreateChatReport`.
Once the chat is fetched/created we can use it to dismiss the settings view.

**Comment by mateocole** [comment]
Alright, to clean this up a little. It sounds like there might be two things being asked here.

1. Reporting a bug is taking you back to the settings and not to concierge 
2. Concierge might not be the best route for reporting a bug

For this specific issue, can this be reorganized to fit scenario 1 as that would be the specific bug?

**Comment by Santhosh-Sellavel** [comment]
> Reporting a bug is taking you back to the settings and not to concierge

Yes, I understood this.

> Concierge might not be the best route for reporting a bug
> For this specific issue, can this be reorganized to fit scenario 1 as that would be the specific bug?

What do you mean here, I'm not following @mateocole

**Comment by parasharrajat** [comment]
First-time navigation is working but the second time there is an issue. Not going back does not seem the root cause of it. Did you find why it is not working for the second time?

**Comment by Santhosh-Sellavel** [comment]
> First-time navigation is working but the second time there is an issue. Not going back does not seem the root cause of it. Did you find why it is not working for the second time?

Yes, of course, We don't allow to navigate to reportId which is already active, that is Concierge.

@parasharrajat 
Also If you are a new user you won't even get navigated for the first time.

**Comment by mateocole** [comment]
> What do you mean here, I'm not following @mateocole

So, the intended action would be to direct you to concierge to report the bug. This would not be relative to this issue as it would be a separate proposal as that would be changing the intended action, if I am understanding the suggestion correctly.

Said differently, I think the focus of the this issue should be to focus on fixing the bug that does not redirect back to concierge.

**Comment by Santhosh-Sellavel** [comment]
If we focus on navigating to Concierge for reporting bugs, then yes, @mateocole my suggestion could be a separate proposal.

**Comment by parasharrajat** [comment]
Yes, But you open the about page again. You are on /settings/about page. and thus navigating to the Report route should work.
> Yes, of course, We don't allow to navigate to reportId which is already active, that is Concierge.

Could you please link me to where are we doing that?

This seems like more of an issue with `Navigation.navigate` doing `Navigation.dismissModal()` sounds like a hack to fix this as we are already navigating to the Report route.

**Comment by Santhosh-Sellavel** [comment]
Here, 
<img width="922" alt="Screenshot 2021-08-14 at 2 49 40 AM" src="https://user-images.githubusercontent.com/85645967/129419501-bd35f246-70ec-437c-8d79-364ae17e3f2b.png">

Calling `Navigation.dismissModal()` was a simple solution suggestion as I said in the proposal, yeah I agree with you completely!

> Calling both Navigation.dismissModal() and Navigation.navigate shows that there is some issue with navigation implementation. 

There is no issue with implementation, cases are handled well.

**Comment by Santhosh-Sellavel** [comment]
@parasharrajat do you get it now?  

Here we can make some kind of exception for `concierge`, we have activeReportID could use that to find whether it's Concierge using `states` or we have to add something in the state to track currently active login something like that.

**Comment by Santhosh-Sellavel** [comment]
dismissingModals will happen anyway abruptly, we could just do it proactively, there's nothing wrong with that! 
But that's not the only solution here!

**Comment by parasharrajat** [comment]
> dismissingModals will happen anyway abruptly, we could just do it proactively, there's nothing wrong with that!
Definitely navigating two times is an issue.

> Navigation.dismissModal() was a simple solution suggestion as I said in the proposal, yeah I agree with you completely it sounds like a hack, but it works without any issues!

Mostly navigate should handle this. Navigate method is adjusted many times to make it work as it does now. You can look into the history of it. So if we found an issue with it my suggestion is we improve it.

> There is no issue with implementation, cases are handled well.

First time it work as it should and we can navigate to `concierge` report but second time it does not. How can you say `There is no issue with implementation, cases are handled well`.

**Comment by Santhosh-Sellavel** [comment]
> Mostly navigate should handle this. Navigate method is adjusted many times to make it work as it does now. You can look into the history of it. So if we found an issue with it my suggestion is we improve it.

Sure @parasharrajat , as i said earlier`dismissModal` It was not the only solution.

Thanks for inputs, I'll look through the history.


What do you think about this 
https://github.com/Expensify/App/issues/4545#issuecomment-898732449

Please post your thoughts this it will be help ful. Thanks

**Comment by parasharrajat** [comment]
> Here we can make some kind of exception for concierge, we have activeReportID could use that to find whether it's Concierge using states or we have to add something in the state to track currently active login something like that.

Sounds very complicated. Not sure how feasible it would be.

**Comment by mateocole** [comment]
@Santhosh-Sellavel any update on solutoins?

**Comment by Santhosh-Sellavel** [comment]
@mateocole  What's your thought on this solution 
https://github.com/Expensify/App/issues/4545#issuecomment-896806869

**Comment by parasharrajat** [comment]
@mateocole Please review.

### Proposal

1. As I explained above that there is an issue with our core Navigate method.
2. We do block refreshing the page when we navigate to the existing report. It helps us remove the rerender of the many heavy components.
3. But we are not handling the navigation correctly while doing it.

To fix it we have to think about the modals and other app architecture. We should pop the top Stack router and then close the modal.
TO do this. we can update the code https://github.com/Expensify/App/blob/083405a60bfb059224565718fef4725e4e4ab665/src/libs/Navigation/CustomActions.js#L27
to
```javascript
   if (activeReportID === params.reportID) {
            if (state.type !== 'drawer') {
                navigationRef.current.dispatch(() => {
                    // If there are multiple routes then we can pop back to the first route
                    if (state.routes.length > 1) {
                        return StackActions.popToTop();
                    }

                    // Otherwise, we are already on the last page of a modal so just do nothing here 
                    // as goBack() will navigate us
                    // back to the screen we were on before we opened the modal.
                    return StackActions.pop(0);
                });
                Navigation.goBack();
            }
            return DrawerActions.closeDrawer();
        }

```
**It fixes the core issue throughout the app**

 Also, this action is doing two things that it should not and we should break this into two actions. I will refactor this CustomAction to two separate actions to correctly reflects what's happening here.

**Comment by Santhosh-Sellavel** [comment]
@parasharrajat Does this solution work mate?

**Comment by Santhosh-Sellavel** [comment]
@parasharrajat I don't know why are you proposing the solution for the issue which I raised & I'm working on it. Also in discussion with @matecole.  

You do have tasks & bugs on hand already! Which I could solve can I propose solutions there? 
Do you think is it fair to do something like that?  Personally, I think is not fair.

If you are really interested in solving it you could have asked first.

**Comment by parasharrajat** [comment]
It's been 10 days since you have proposed a solution that I thought is not the best solution and I mentioned the same to you. I also continued discussion with you to help you find another instead of proposing And after being asked for a proposal you replied here https://github.com/Expensify/App/issues/4545#issuecomment-901835342 which was the same so I proposed a solution.

>  I'm working on it. 

Please don't start on an issue until your proposal is accepted. 

> You do have tasks & bugs on hand already! Which I could solve can I propose solutions there?
> Do you think is it fair to do something like that? Personally, I think is not fair.

You always have first priority over others if you raise an issue. it's on the Engineer to decide who is going to work on a particular issue. Anyone can apply on an issue.

There are several contributors who are working here to find solutions to many problems but only one is going to be selected to solve that at last. All of them have contributed their time and efforts to propose a fix. Is it fair to them if others are selected over them?

> You do have tasks & bugs on hand already! 
I was not working on anything when I proposed the solution.

You can use the solution if this is not fair to you and I take full responsibility for this discussion if I was not fair to you and I didn't tell you that I don't feel that your solution is good or If I went against the contribution policy.

**Comment by marcaaron** [comment]
Hey @Santhosh-Sellavel,  This could probably be a bit clearer in our [contributing guidelines](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#proposing-a-job-that-expensify-hasnt-posted), but I just want to reiterate that there's **no rule against someone proposing a solution to an issue created by another**. Regardless, I think there has been some confusion here and your assignment to the issue was most likely accidental. Contributors are only assigned to issues once a job is created and an Expensify engineer reviews the proposal. We'll try to do better next time! Thanks :)

**Comment by marcaaron** [comment]
@mateocole feel free to pass this over to `Engineering` or slap the `External` label if the issue looks good to ya.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @MitchExpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MitchExpensify** [comment]
Was my unassigne-ment a mistake @mateocole ?

**Comment by mateocole** [comment]
@MitchExpensify Sorry about that, I have no idea why I unassigned you on this? I think I meant to unassign myself perhaps?

**Comment by AndrewGable** [comment]
@parasharrajat - Is your proposal here: https://github.com/Expensify/App/issues/4545#issuecomment-901944215 up to date? Would you like to work on this issue? I can review it if so.

**Comment by parasharrajat** [comment]
Yes that is up-to-date.

**Comment by AndrewGable** [comment]
@parasharrajat - Sounds good. I've looked into the issue a bit more and agree with your solution. @MitchExpensify can you hire @parasharrajat please? Thanks.

**Comment by MitchExpensify** [comment]
Hire offer sent @AndrewGable!

**Comment by Santhosh-Sellavel** [comment]
@MitchExpensify or @AndrewGable 

As posted in #expensify-open-source [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1630447493236300)

> This change is in effect as of yesterday and the bonus for independently reporting an issue is only valid from yesterday forward. If there are existing, open, GH issues that you believe you should be compensated for reporting a bug, please comment on the issue.

I believe I should be compensated for reporting this issue. Thanks!

**Comment by MitchExpensify** [comment]
Thanks for the catch @Santhosh-Sellavel ! Offer sent so I can pay you for the reporting of the issue

**Comment by kadiealexander** [comment]
Hi @parasharrajat, I've placed this issue on hold as per [this update](https://expensify.slack.com/archives/C01GTK53T8Q/p1631754482374800), as we are prioritising issues related to a feature release scheduled for 9/24. As an apology for the delay, we will add a $100 bonus as a thank you for waiting.

**Comment by MitchExpensify** [comment]
Not overdue

**Comment by MitchExpensify** [comment]
On hold

**Comment by MitchExpensify** [comment]
Not overdue, awaiting payment

**Comment by mallenexpensify** [comment]
Paid @Santhosh-Sellavel $250 for reporting
Paid @parasharrajat, inc. $250 for `n6-hold` 
Thanks gents, closing
